.github/workflows/ci.yml

```diff
@@ -279,3 +281,22 @@ jobs:
 
             If this is not what was expected, commit changes to `config/versions.json`.
             </details>
+
+      - name: PR Description Notifier
+        env:
+          GH_TOKEN: ${{ github.token }}
+          PR_BODY_PATH: ./body.txt
+          PR_BODY_PATH_UPDATED: ./updated.body.txt
+          PRERELEASE_SECTION_REGEX: "^:rocket: .* :rocket:$"
+          PRERELEASE_SECTION: ":rocket: The latest prerelease `${{ needs.defaults.outputs.root-sbd }}/${{ needs.check-spack-yaml.outputs.prerelease }}` is here: ${{ steps.comment.outputs.comment-link }} :rocket:"
+        run: |
+          gh pr view ${{ github.event.pull_request.number }} --repo ${{ github.repository }} --json body --jq .body > ${{ env.PR_BODY_PATH }}
+
+          if grep -q '${{ env.PRERELEASE_SECTION_REGEX }}' ${{ env.PR_BODY_PATH }}; then  # there is an existing prerelease section
+            # Replace the existing prerelease section
+            awk '{gsub(/${{ env.PRERELEASE_SECTION_REGEX }}/, "${{ env.PRERELEASE_SECTION }}")}1' ${{ env.PR_BODY_PATH }} > ${{ env.PR_BODY_PATH_UPDATED }}
```

Comment:
This solution with `awk` seems a bit more cryptic to me, but it works.

It also requires you to have another file (even though I think you could output the result to the same `${{ env.PR_BODY_PATH }}` file), and it doesn't solve all issues with metacharacters etc., but with the current structure of `PRERELEASE_SECTION` should work fine.

Is the `PRERELEASE_SECTION` structure supposed to stay the same? Because in that case I wonder if it might be better to only substitute the needed portions (which I believe are `${{ needs.defaults.outputs.root-sbd }}/${{ needs.check-spack-yaml.outputs.prerelease }}` and `${{ steps.comment.outputs.comment-link }}`).

Anyway it's a matter of taste, I think there shouldn't be real issues with the current implementation.

Reply:
Yes `awk` is a bit more cryptic, but the substitution string crucially is not a regex string (like in `sed`), just a regular one. Hence we could put whatever we wanted into it, metacharacters included. 
When I was looking into it, I hadn't seen anyone write to the file they were reading, only writing to a temp file and then `mv`ing it. 
The `PRERELEASE_SECTION` may change, who knows - having it in one place (the `env` var) is preferable to updating it in multiple.

Reply:
Makes sense.